zerver/lib/typed_endpoint_validators.py

```diff
@@ -21,18 +22,26 @@ def check_string_fixed_length(string: str, length: int) -> Optional[str]:
     return string
 
 
-def check_string_in(val: str, possible_values: List[str]) -> str:
+def check_string_in(val: str, possible_values: Iterable[str]) -> str:
     if val not in possible_values:
         raise ValueError(_("Not in the list of possible values"))
     return val
 
 
-def check_int_in(val: int, possible_values: List[int]) -> int:
+def check_int_in(val: int, possible_values: Iterable[int]) -> int:
     if val not in possible_values:
         raise ValueError(_("Not in the list of possible values"))
     return val
 
 
+def check_int_in_validator(possible_values: Iterable[int]) -> AfterValidator:
```

Comment:
Don’t use `Iterable`; that would allow passing an iterator that would be iterated once and exhausted, leading to silent misbehavior. Consider `Collection` instead.